### ISSUE TITLE
TemporalMonthDayToString and TemporalYearMonthToString: Don't output full date when `calendarName` option is "never"?

### ISSUE THREAD

**Issue by anba** [tc39/proposal-temporal#2640]
```js
let cal = new Proxy({id: "cal"}, { has(t, pk) { return true; } });
let pmd = new Temporal.PlainMonthDay(8, 1, cal);
console.log(pmd.toString({calendarName: "never"}));
```

currently prints `"1972-08-01"`, but maybe should instead print `"08-01"`?

```js
let cal = new Proxy({id: "cal"}, { has(t, pk) { return true; } });
let pym = new Temporal.PlainYearMonth(2023, 8, cal);
console.log(pym.toString({calendarName: "never"}));
```

currently prints `"2023-08-01"`, but maybe should instead print `"2023-08"`?

The full date string is computed when the calendar identifier isn't `iso8601`, but when then `calendarName` option is `"never"`, it's probably not too useful to return the full date string, because the calendar identifier isn't part of the output, so returning the reference year resp. day doesn't provide any additional context.

**Comment by justingrant** [comment]
Meeting 2023-08-17: 
* We should never output a non-ISO calendar PYM/PMD without the reference date, because the result would be misleading (and not round-trippable) for all calendars except "gregory".
* We agreed that `{calendarName: "never"}` is not useful in this case (and will return a different format for the ISO calendar vs. other calendars) but we also didn't see a strong enough justification to, at this late date, make a normative change to remove `{calendarName: "never"}` for these two types. And there was some support for keeping `toString` options consistent across types. 

TL;DR - we decided to leave the current behavior as-is.

FYI @sffc in case you have concerns with the decision above.
